conans/model/conan_file.py

```diff
@@ -10,44 +10,205 @@
+    requires = None
+
+    #: List or tuple of strings for dependencies. Represents a build tool like "cmake". If there is
+    #: an existing pre-compiled binary for the current package, the binaries for the tool_require
+    #: won't be retrieved. They cannot conflict.
+    tool_requires = None
+
+    #: List or tuple of strings for dependencies. Generic type of build dependencies that are not
+    #: applications (nothing runs), like build scripts. If there is
+    #: an existing pre-compiled binary for the current package, the binaries for the build_require
+    #: won't be retrieved. They cannot conflict.
+    build_requires = None
+
+    #: List or tuple of strings for dependencies in the host context only. Represents a test tool
+    #: like "gtest". Used when the current package is built from sources.
+    #: They don't propagate information to the downstream consumers. If there is
+    #: an existing pre-compiled binary for the current package, the binaries for the test_require
+    #: won't be retrieved. They cannot conflict.
+    test_requires = None
+
+    #: The attribute ``no_copy_source`` tells the recipe that the source code will not be copied from
+    #: the ``source_folder`` to the ``build_folder``. This is mostly an optimization for packages
+    #: with large source codebases or header-only, to avoid extra copies.
+    no_copy_source = False
+
+    #: The folder where the recipe *conanfile.py* is stored, either in the local folder or in
+    #: the cache. This is useful in order to access files that are exported along with the recipe,
+    #: or the origin folder when exporting files in ``export(self)`` and ``export_sources(self)``
+    #: methods.
+    recipe_folder = None
+
+    #: Object storing all the information needed by the consumers
+    #: of a package: include directories, library names, library paths... Both for editable
+    #: and regular packages in the cache. It is only available at the ``layout()`` method.
+    #:
+    #:  - ``self.cpp.package``: For a regular package being used from the Conan cache. Same as
+    #:    declaring ``self.cpp_info`` at the ``package_info()`` method.
+    #:  - ``self.cpp.source``: For "editable" packages, to describe the artifacts under
+    #:    ``self.source_folder``
+    #:  - ``self.cpp.build``: For "editable" packages, to describe the artifacts under
+    #:    ``self.build_folder``.
+    #:
+    cpp = None
+
+    #: For the dependant recipes, the declared environment variables will be present during the
+    #: build process. Should be only filled in the ``package_info()`` method.
+    #: It will be initialized in the ``__init__()`` with an ``Environment()`` object.
+    buildenv_info = None
+
+    #: For the dependant recipes, the declared environment variables will be present at runtime.
+    #: Should be only filled in the ``package_info()`` method.
+    #: It will be initialized in the ``__init__()`` with an ``Environment()`` object.
```

Comment:
This reads a bit too internal implementation, but not sure how to explain it better.